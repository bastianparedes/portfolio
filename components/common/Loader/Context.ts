import { useState } from 'react';

import constate from 'constate';

interface ContextLoaderPropsType {
  initialLoaderCounter?: number;
}

interface ContextLoaderReturnType {
  loaderCounter: number;
  setLoaderCounter: (number: number) => void;
  addLoaderCounter: () => void;
  substractLoaderCounter: () => void;
}

function useContext({
  initialLoaderCounter = 0
}: ContextLoaderPropsType): ContextLoaderReturnType {
  const [loaderCounter, setLoaderCounter] = useState(initialLoaderCounter);

  const addLoaderCounter = (): void =>
    setLoaderCounter((counter) => counter + 1);
  const substractLoaderCounter = (): void =>
    setLoaderCounter((counter) => counter - 1);

  return {
    addLoaderCounter,
    loaderCounter,
    setLoaderCounter,
    substractLoaderCounter
  };
}

const [LoaderProvider, useLoaderContext] = constate(useContext);
export { LoaderProvider, useLoaderContext };
