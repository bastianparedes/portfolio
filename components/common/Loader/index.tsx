import React from 'react';

import Shadow from '../Shadow';
import styles from './styles.module.scss';

const Loader = (): JSX.Element => {
  return (
    <Shadow modes={['blur']}>
      <div className={styles.wrapper}>
        <div className={styles.boxWrap}>
          <div className={`${styles.box} ${styles.one}`} />
          <div className={`${styles.box} ${styles.two}`} />
          <div className={`${styles.box} ${styles.three}`} />
          <div className={`${styles.box} ${styles.four}`} />
          <div className={`${styles.box} ${styles.five}`} />
          <div className={`${styles.box} ${styles.six}`} />
        </div>
      </div>
    </Shadow>
  );
};

export default Loader;
