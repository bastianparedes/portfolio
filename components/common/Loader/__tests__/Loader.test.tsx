import React from 'react';
import { render } from '@testing-library/react';
import Loader from '..';

jest.mock(
  '../../Shadow',
  () =>
    ({ children }: { children: React.ReactNode }) =>
      <>{children}</>
);

describe('<Loader />', () => {
  it('should render', () => {
    const { container } = render(<Loader />);
    expect(container).toMatchSnapshot();
  });
});
