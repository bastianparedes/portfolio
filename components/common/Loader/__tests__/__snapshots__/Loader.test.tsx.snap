// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="boxWrap"
    >
      <div
        class="box one"
      />
      <div
        class="box two"
      />
      <div
        class="box three"
      />
      <div
        class="box four"
      />
      <div
        class="box five"
      />
      <div
        class="box six"
      />
    </div>
  </div>
</div>
`;
